root/edit/details/EditAlias.js

```diff
@@ -0,0 +1,109 @@
+// @flow
+
+import React from 'react';
+import _ from 'lodash';
+
+import formatEntityTypeName from '../../static/scripts/common/utility/formatEntityTypeName';
+import bracketed from '../../static/scripts/common/utility/bracketed';
+import DescriptiveLink from '../../static/scripts/common/components/DescriptiveLink';
+import EntityLink from '../../static/scripts/common/components/EntityLink';
+import isolateText from '../../static/scripts/common/utility/isolateText';
+import Diff from '../../static/scripts/edit/components/edit/Diff';
+import locales from '../../static/scripts/common/constants/locales';
+import FullChangeDiff from '../../static/scripts/edit/components/edit/FullChangeDiff';
+import formatDate from '../../static/scripts/common/utility/formatDate';
+import yesNo from '../../static/scripts/common/utility/yesNo';
+
+type CompT<T> = {
+  new: T,
+  old: T,
+};
+
+type OnlyNameT = {
+  name: string,
+} | null;
+
+type DisplayDataT = {
+  alias: CompT<string>,
+  artist?: OnlyNameT,
```

Comment:
Well, it's likely related to the `const entity = display[entityType];` line below, but it doesn't really help type that properly (and we don't list any other possible entity types here). So it should be removed, or the edit type type should have a parameter for the entity type.

Reply:
Yea artist is there because of the display[entityType].

Reply:
I'm removing the line because of ambiguity in the various options that could be the entity. Place alias, artist alias, etc. Let me know if there is an alternative